cli/repl.rs

```diff
@@ -1,73 +1,78 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
 
-use crate::deno_dir::DenoDir;
+use crate::global_state::GlobalState;
+use crate::inspector::InspectorSession;
 use deno_core::error::AnyError;
+use deno_core::serde_json::json;
+use rustyline::error::ReadlineError;
 use rustyline::Editor;
-use std::fs;
-use std::path::PathBuf;
 
-pub struct Repl {
-  editor: Editor<()>,
-  history_file: PathBuf,
-}
+pub async fn run(
+  _global_state: &GlobalState,
+  mut session: Box<InspectorSession>,
+) -> Result<(), AnyError> {
+  // Our inspector is unable to default to the default context id so we have to specify it here.
+  let context_id: u32 = 1;
 
-impl Repl {
-  pub fn new(history_file: PathBuf) -> Self {
-    let mut repl = Self {
-      editor: Editor::<()>::new(),
-      history_file,
-    };
+  session
+    .post_message("Runtime.enable".to_string(), None)
+    .await?;
 
-    repl.load_history();
-    repl
-  }
+  let mut editor = Editor::<()>::new();
 
-  fn load_history(&mut self) {
-    debug!("Loading REPL history: {:?}", self.history_file);
-    self
-      .editor
-      .load_history(&self.history_file.to_str().unwrap())
-      .map_err(|e| {
-        debug!("Unable to load history file: {:?} {}", self.history_file, e)
-      })
-      // ignore this error (e.g. it occurs on first load)
-      .unwrap_or(())
-  }
+  println!("Deno {}", crate::version::DENO);
+  println!("exit using ctrl+d or close()");
 
-  fn save_history(&mut self) -> Result<(), AnyError> {
-    fs::create_dir_all(self.history_file.parent().unwrap())?;
-    self
-      .editor
-      .save_history(&self.history_file.to_str().unwrap())
-      .map(|_| debug!("Saved REPL history to: {:?}", self.history_file))
-      .map_err(|e| {
-        eprintln!("Unable to save REPL history: {:?} {}", self.history_file, e);
-        e.into()
-      })
-  }
+  loop {
+    let line = editor.readline("> ");
```

Comment:
Spawn blocking, most likely.

Reply:
Yes, otherwise event loop won't run when waiting on the input